tools/github_readme_sync/readme.py

```diff
@@ -253,21 +270,43 @@ def get_stable_version(self) -> str:
 
         raise ValueError("No stable version found")
 
-    def caption_markdown_images(self, markdown_text: str) -> str:
+    def parse_images(self, markdown_text: str) -> str:
         def replace_image(match):
             if any(ignore_image in match.groups()[1] for ignore_image in IGNORE_IMAGES):
                 return match.group(0)
             alt_text, image_src = match.groups()
+
+            # Split image source and fragment
+            src_parts = image_src.split("#")
+            clean_src = src_parts[0]
+            style = "border-radius: 8px;"
+
+            # Parse and sanitize style parameters from fragment
```

Comment:
**issue (blocking)**: I would like to see two things that reinforce each other: a sanitization library and an allowlist of supported keys.

Please look at the `nh3` sanitization library to sanitize input: https://nh3.readthedocs.io/en/latest/#. I don't want us to maintain our own sanitization code. I would rather rely on the expertise of people who care about this sort of thing.

After the CSS keys are extracted, they should be checked against an allowlist of keys, in this case, consisting of a single one: `width`. However, I noticed `height` in the tests. Whatever the list, I think we should enumerate an allowlist to vastly reduce what can be put into the HTML.

One way to accomplish the above would be to first filter `key, values` for the allowlist of keys, i.e., `width`. Then, second, naively and unsafely construct the final HTML `<figure><img...` and then put the final HTML through the `nh3.clean()` before returning it.